File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.cs
using System;

public class Solution {
    public int[] solution(int[] arr, int n) {
        int[] answer = new int[arr.Length];
        for(int i = 0 ;i<arr.Length;i++)
        {
            if(arr.Length % 2 == 0)
            {
                if(i % 2 == 0)
                    answer[i] = arr[i];
                else
                    answer[i] = arr[i] + n;
            }
            else
            {
                if(i % 2 == 0)
                    answer[i] = arr[i] + n;
                else
                    answer[i] = arr[i];
            }
        }
        return answer;
    }
}

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, int n) {
    vector<int> answer(arr);
    int i = arr.size() % 2 != 0 ? 0 : 1;
    for(; i < arr.size(); i = i + 2)
        answer[i] = answer[i] + n;
    return answer;
}